charts/dynatrace-oneagent-operator/v0.8.1/questions.yml

```diff
@@ -27,7 +27,7 @@ questions:
   description: "Your Dynatrace API token - You can generate this token in your Dynatrace environment"
   default: ""
   type: string
-  required: true
+  required: false
```

Comment:
Please remove the required field altogether, false is the default behavior

Reply:
Done!

Reply:
I test installed the chart and the following error will happen if this field is left blank.

`Failed to install app dynatrace-oneagent-operator. Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: unknown object type "nil" in Secret.data.apiToken; waiting on helm-controller_c-nk79q`

Reply:
This is not required only if you provide the values.yaml in a special way (mode: apm)
However, to prevent misunderstandings and a broken chart I re-added the required: false flag again.

Reply:
Adding the `required: false` will produce the same error because adding `required: false` is the same as not including a required. The error happens because leaving this field empty in the UI results in `secret.apiToken` getting assigned an empty string from `default: ""` and eventually at object creation `Secret.data.apiToken` gets this value too and something doesn't like it.

Reply:
It shouldn't be a problem. The only place where we use the secret.apiToken is in secret.yaml
There we have a dedicated check if the mode is set to fullstack. If so we  add it.
See https://github.com/rancher/helm3-charts/pull/24/files#diff-fb5876da0aa5246c079da3d9c3b52efeR23

Reply:
I understand, but although that error is not a problem for the chart itself, it does bubble up to Rancher and causes the installation to fail. An user will then need to remove the failed installation and retry installing it once again. So in a way, they are forced to provide a `secret.apiToken` if they want to successfully install the chart.

Reply:
Talked about this with my colleague - seems like helm is not flexible enough to satisfy our needs here.
Furthermore the APItoken will be required in one of the next releases anyway. Therefore I did set it to 'Required: true'